recipes/ccfits/all/patches/0001-adapt-cmakelists-for-conan-2.6.patch

```diff
@@ -0,0 +1,47 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,8 +1,6 @@
+ PROJECT(CCfits)
+-CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
++CMAKE_MINIMUM_REQUIRED(VERSION 3.8)
+ 
+-set (CMAKE_CXX_STANDARD_REQUIRED on)
+-set (CMAKE_CXX_STANDARD 11)
+ 
+ # Allow the developer to select whether to build Dynamic or Static libraries
+ OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
+@@ -25,8 +23,7 @@ ENDIF()
+ 
+ #add_subdirectory (src)
+ 
+-find_package(CFITSIO REQUIRED)
+-INCLUDE_DIRECTORIES(${CFITSIO_INCLUDE_DIR})
++
+ 
+ SET (LIB_TYPE STATIC)
+ IF (BUILD_SHARED_LIBS)
+@@ -58,15 +55,17 @@ SET(SRC_FILES
+ )
+ 
+ ADD_LIBRARY(${LIB_NAME} ${LIB_TYPE} ${H_FILES} ${SRC_FILES})
+-TARGET_LINK_LIBRARIES(${LIB_NAME} ${CFITSIO_LIBRARY}
```

Comment:
cfitsio is in cci and looks good, what exactly stops ccfits to find it using `find_package`?

Reply:
There is no official Find or config file of cfitsio, so it would be fragile (upstream could decide to create one in the future, we may have to change CMake name in cfitsio recipe, then ccfits recipe would break).

Reply:
But I've sent an email to ccfits authors, with most of this patch.

Reply:
Checked the FindCFITSIO, so yes, it's really better to use CONAN_LIBS directly.

Reply:
why doesn't it handle transitive dependencies? CMake is usually able to propagate compiler/linker flags properly across transitive dependencies

Reply:
With targets only, if it's properly defined, and if you link to those targets. Read upstream FindCFITSIO.cmake, what it searches, what it defines, and how it is then linked to ccfits.
Don't trust too much CMake black magic.

Reply:
but if we use conan to build, then `FindCFITSIO.cmake` should be generated by conan generator, right?
and in this case, it should be able to propagate transitive flags

Reply:
I've explained why it can't work: https://github.com/conan-io/conan-center-index/pull/6611#discussion_r680479586

Reply:
it sounds like a very unfortunate situation...
it means, for the specific project, we should generate a specific find module with specific variables?
I mean, recipe for `cfitsio` shouldn't contain that:
```
self.cpp_info.names["cmake_find_package"] = "CFITSIO"
```
but instead, recipe for `ccfits` should make a "bridge" `FindCFITSIO.cmake` with unique CMake variables (like `CFITSIO_LIBRARY`).
otherwise, heavy patching should be applied

Reply:
The only reliable solution outside of conan for `ccfits` is not use official `cfitsio.pc` file.